Uniswap/interface
# feat: new app provider w/ fallback behavior

No UX change.

This should make it possible to fall back from a failing rpc url to one that is more stable. 

To test:
- enable fallback_provider feature flag
![image](https://github.com/Uniswap/interface/assets/5773490/f98d7ef1-2f83-490a-863f-125c3cc54376)
- view mainnet
- do not connect a wallet
- observe successful rpc calls
- block infura (or quiknode) in dev tools
- observe rpc requests routed to unblocked rpc

You may also pull this down locally and update the default update interval to a very small value (1000 ms) and use devtools network blocker to alternatively disable quicknode/infura.
